packages/react/src/editor/BlockNoteView.tsx

```diff
@@ -161,31 +148,57 @@ function BlockNoteViewComponent<
     [editor, portalManager]
   );
 
+  // The BlockNoteContext makes sure the editor and some helper methods
+  // are always available to nesteed compoenents
+  const blockNoteContext: BlockNoteContextValue<any, any, any> = useMemo(() => {
+    return {
+      ...existingContext,
+      editor,
+      setContentEditableProps,
+    };
+  }, [existingContext, editor]);
+
+  // We set defaultUIProps and editorProps on a different context, the BlockNoteViewContext.
+  // This BlockNoteViewContext is used to render the editor and the default UI.
+  const defaultUIProps = {
+    formattingToolbar,
+    linkToolbar,
+    slashMenu,
+    emojiPicker,
+    sideMenu,
+    filePanel,
+    tableHandles,
+  };
+
+  const editorProps = {
+    autoFocus,
+    className,
+    editorColorScheme,
+    mount,
+    contentEditableProps,
+    ref,
+    ...rest,
+  };
+
   return (
-    <BlockNoteContext.Provider value={context as any}>
-      <ElementRenderer ref={setElementRenderer} />
-      {!editor.headless && (
-        <>
-          <Portals contentComponent={portalManager} />
-          <div
-            className={mergeCSSClasses(
-              "bn-container",
-              editorColorScheme || "",
-              className || ""
+    <BlockNoteContext.Provider value={blockNoteContext}>
+      <BlockNoteViewContext.Provider
+        value={{
+          editorProps,
+          defaultUIProps,
+        }}>
+        <ElementRenderer ref={setElementRenderer} />
+        {!editor.headless && (
+          <>
+            <Portals contentComponent={portalManager} />
```

Comment:
Feels like the portals should just be part of `BlockNoteViewEditor` to clean this up

Reply:
see update